src/Microsoft.Graph.Cli.Core/Commands/Authentication/LoginCommand.cs

```diff
@@ -45,14 +54,24 @@ public Command Build()
             string[] scopes = context.ParseResult.GetValueForOption(scopesOption) ?? new string[] { };
             var clientId = context.ParseResult.GetValueForOption(clientIdOption);
             var tenantId = context.ParseResult.GetValueForOption(tenantIdOption);
+            var certificateName = context.ParseResult.GetValueForOption(certificateNameOption);
+            var certificateFilePath = context.ParseResult.GetValueForOption(certificatePathOption)?.FullName;
+            var certificateThumbPrint = context.ParseResult.GetValueForOption(certificateThumbPrintOption);
             var strategy = context.ParseResult.GetValueForOption(strategyOption);
             var host = context.BindingContext.GetRequiredService<IHost>();
             var cancellationToken = context.BindingContext.GetRequiredService<CancellationToken>();
 
             var authUtil = host.Services.GetRequiredService<IAuthenticationCacheUtility>();
-            var authService = await this.authenticationServiceFactory.GetAuthenticationServiceAsync(strategy, tenantId, clientId, cancellationToken);
+            string? password = null;
+            if (!string.IsNullOrWhiteSpace(certificateFilePath) && File.Exists(certificateFilePath))
```

Comment:
shouldn't we try to open it first before asking a password? maybe it's not password protected.